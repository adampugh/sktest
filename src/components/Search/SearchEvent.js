import React from 'react';
import { Link } from 'react-router-dom';
import { FaCalendarAlt, FaMapMarkerAlt } from 'react-icons/fa';


const SearchEvent = ({ name, date, desc, img, venue, id }) => (
    <div className="searchEvent">
        <img src={img} alt={name} />
        <div className="searchEvent__text">
            <h2 className="title">{name}</h2>
            <p className="searchEvent__text__desc">{desc}</p>
            <div className="searchEvent__text__date">
                <FaCalendarAlt />
                <p>{date}</p>
            </div>
            <div className="searchEvent__text__venue">
                <FaMapMarkerAlt />
                <p>{venue}</p>
            </div>
            <Link to={`/event/${id}`}>
                <button className="btn">view details</button>
            </Link>
        </div>
    </div>
);

export default SearchEvent;