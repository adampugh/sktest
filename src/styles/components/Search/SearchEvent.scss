.searchEvent {
    background: white;
    text-align: center;
    max-width: 100%;
    @include boxshadow;
    h2 {
        color: #121212;
        margin-bottom: 10px;
    }
    img {
        width: 100%;
        max-width: 100%;
    }
    
}

.searchEvent__text__date, .searchEvent__text__venue {
    text-align: left;
    display: block;
    margin: 10px 0;
    p {
        display: inline;
        margin-left: 10px;
    }
}

.searchEvent__text__desc {
    height: 100px;
}

.searchEvent__text {
    padding: 20px;
    button {
        margin-top: 10px;
        &:hover {
            @include boxshadow;
            cursor: pointer;
        }
    }
}