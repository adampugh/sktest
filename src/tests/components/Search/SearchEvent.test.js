import React from 'react';
import '../../setupTests';
import { shallow } from 'enzyme';

import SearchEvent from '../../../components/Search/SearchEvent';

describe("<SearchEvent />", () => {
    it("should match snapshot", () => {
        const props = {
            name: '123', date: '123', desc: '123', img: '123', venue: '123', id: '123'
        }

        const wrapper = shallow(<SearchEvent {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});