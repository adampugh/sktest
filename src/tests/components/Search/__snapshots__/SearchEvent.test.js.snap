// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchEvent /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchEvent
    date="123"
    desc="123"
    id="123"
    img="123"
    name="123"
    venue="123"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="123"
          src="123"
        />,
        <div
          className="searchEvent__text"
        >
          <h2
            className="title"
          >
            123
          </h2>
          <p
            className="searchEvent__text__desc"
          >
            123
          </p>
          <div
            className="searchEvent__text__date"
          >
            <FaCalendarAlt />
            <p>
              123
            </p>
          </div>
          <div
            className="searchEvent__text__venue"
          >
            <FaMapMarkerAlt />
            <p>
              123
            </p>
          </div>
          <Link
            to="/event/123"
          >
            <button
              className="btn"
            >
              view details
            </button>
          </Link>
        </div>,
      ],
      "className": "searchEvent",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "123",
          "src": "123",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="title"
            >
              123
            </h2>,
            <p
              className="searchEvent__text__desc"
            >
              123
            </p>,
            <div
              className="searchEvent__text__date"
            >
              <FaCalendarAlt />
              <p>
                123
              </p>
            </div>,
            <div
              className="searchEvent__text__venue"
            >
              <FaMapMarkerAlt />
              <p>
                123
              </p>
            </div>,
            <Link
              to="/event/123"
            >
              <button
                className="btn"
              >
                view details
              </button>
            </Link>,
          ],
          "className": "searchEvent__text",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "123",
              "className": "title",
            },
            "ref": null,
            "rendered": "123",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "123",
              "className": "searchEvent__text__desc",
            },
            "ref": null,
            "rendered": "123",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <FaCalendarAlt />,
                <p>
                  123
                </p>,
              ],
              "className": "searchEvent__text__date",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "123",
                },
                "ref": null,
                "rendered": "123",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <FaMapMarkerAlt />,
                <p>
                  123
                </p>,
              ],
              "className": "searchEvent__text__venue",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "123",
                },
                "ref": null,
                "rendered": "123",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="btn"
              >
                view details
              </button>,
              "to": "/event/123",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "view details",
                "className": "btn",
              },
              "ref": null,
              "rendered": "view details",
              "type": "button",
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="123"
            src="123"
          />,
          <div
            className="searchEvent__text"
          >
            <h2
              className="title"
            >
              123
            </h2>
            <p
              className="searchEvent__text__desc"
            >
              123
            </p>
            <div
              className="searchEvent__text__date"
            >
              <FaCalendarAlt />
              <p>
                123
              </p>
            </div>
            <div
              className="searchEvent__text__venue"
            >
              <FaMapMarkerAlt />
              <p>
                123
              </p>
            </div>
            <Link
              to="/event/123"
            >
              <button
                className="btn"
              >
                view details
              </button>
            </Link>
          </div>,
        ],
        "className": "searchEvent",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "123",
            "src": "123",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="title"
              >
                123
              </h2>,
              <p
                className="searchEvent__text__desc"
              >
                123
              </p>,
              <div
                className="searchEvent__text__date"
              >
                <FaCalendarAlt />
                <p>
                  123
                </p>
              </div>,
              <div
                className="searchEvent__text__venue"
              >
                <FaMapMarkerAlt />
                <p>
                  123
                </p>
              </div>,
              <Link
                to="/event/123"
              >
                <button
                  className="btn"
                >
                  view details
                </button>
              </Link>,
            ],
            "className": "searchEvent__text",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "123",
                "className": "title",
              },
              "ref": null,
              "rendered": "123",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "123",
                "className": "searchEvent__text__desc",
              },
              "ref": null,
              "rendered": "123",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <FaCalendarAlt />,
                  <p>
                    123
                  </p>,
                ],
                "className": "searchEvent__text__date",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "123",
                  },
                  "ref": null,
                  "rendered": "123",
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <FaMapMarkerAlt />,
                  <p>
                    123
                  </p>,
                ],
                "className": "searchEvent__text__venue",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "123",
                  },
                  "ref": null,
                  "rendered": "123",
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="btn"
                >
                  view details
                </button>,
                "to": "/event/123",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "view details",
                  "className": "btn",
                },
                "ref": null,
                "rendered": "view details",
                "type": "button",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
